sparta/sparta/resources/Queue.hpp

```diff
@@ -404,8 +398,14 @@ namespace sparta
               InstrumentationNode::visibility_t stat_vis_avg     = InstrumentationNode::AUTO_VISIBILITY) :
             num_entries_(num_entries),
             vector_size_(nextPowerOfTwo_(num_entries*2)),
+
             name_(name),
-            collector_(nullptr)
+            collector_(nullptr),
+
+            // Make the queue twice as large and a power of two to
+            // allow a complete invalidation followed by a complete
+            // population
+            queue_data_((value_type *)malloc(sizeof(value_type) * vector_size_))
```

Comment:
So what issues did you run into using `new`?

Reply:
We definitely don't want to use new operator here because we don't want initialized memory. We want uninitialized memory so that we can placement-new construct as we get data.

Reply:
Actually, I was curious why not use `new char[]` with the `unique_ptr` that was there before, but ensure the pointer is aligned to a `DataT` address.  I have no problem, really with the malloc, except for the case where the Queue constructor throws, which there will be a memory leak...

Reply:
I did think of using ```std::aligned_storage``` for this.
I will give it a shot.

Reply:
Went along a different route.
Used ```unique_ptr``` with ```malloc``` and custom deleter which does ```free```.

Reply:
Does this mean that value_type no longer needs a default constructor?

Reply:
I believe so, yes.

Reply:
In my test, I just removed the default ctor for value_type and works as expected.